tycho-common/src/simulation/protocol_sim.rs

```diff
@@ -106,22 +106,25 @@ pub trait ProtocolSim: fmt::Debug + Send + Sync + 'static {
     /// Returns the fee of the protocol as ratio
     ///
     /// E.g. if the fee is 1%, the value returned would be 0.01.
+    ///
+    /// # Panics
+    ///
+    /// Panic for protocols with asymmetric fees (e.g. Rocketpool, Uniswap V4 with hooks),
+    /// where a single fee value cannot represent the protocol's fee structure.
     fn fee(&self) -> f64;
 
-    /// Returns the protocol's current spot price of two tokens
-    ///
-    /// Currency pairs are meant to be compared against one another in
-    /// order to understand how much of the quote currency is required
-    /// to buy one unit of the base currency.
+    /// Returns the amount of the quote token required to buy one unit of base token,
+    /// accounting for protocol fees.
     ///
-    /// E.g. if ETH/USD is trading at 1000, we need 1000 USD (quote)
-    /// to buy 1 ETH (base currency).
+    /// E.g. if BTC/USDT is trading at 1000 with a 20% fee, you would need 1,250 USDT (1000 / 0.8)
+    /// to buy 1 BTC (base currency) as you will be deducted 20% of your quote token when buying.
+    /// Hence, the spot price returned would be 1250 USDT per BTC.
```

Comment:
I think the math is pretty self explanatory. We don't need to be too verbose if it's not necessary.
```suggestion
    /// E.g. if BTC/USDT is trading at 1000 with a 20% fee, you would need 1,250 USDT (1000 / 0.8)
    /// to buy 1 BTC (base currency).
```